app/src/main/res/layout/activity_donor_history.xml

```diff
@@ -0,0 +1,172 @@
+    tools:context=".donate.DonorHistoryActivity">
+
+    <com.google.android.material.appbar.MaterialToolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="?attr/actionBarSize"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <TextView
+            style="@style/H3.Button"
+            android:id="@+id/saveButton"
+            android:layout_width="wrap_content"
+            android:layout_height="match_parent"
+            android:layout_gravity="end"
+            android:paddingHorizontal="16dp"
+            android:text="@string/donor_history_save_button"
+            android:textColor="?attr/progressive_color"
+            android:gravity="center"
+            android:clickable="true"
+            android:focusable="true"
+            android:background="?android:attr/selectableItemBackground"/>
+
+    </com.google.android.material.appbar.MaterialToolbar>
+
+    <ScrollView
+        android:id="@+id/donorHistoryScrollView"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:layout_marginBottom="16dp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/toolbar"
+        app:layout_constraintBottom_toTopOf="@id/experimentLink">
+
+        <LinearLayout
+            android:id="@+id/donorStatusContainer"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="vertical">
+
+            <TextView
+                android:id="@+id/donorStatus"
+                style="@style/P"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:text="@string/donor_history_update_donor_status_default"
+                android:textColor="?attr/placeholder_color"
+                android:drawablePadding="16dp"
+                android:paddingVertical="16dp"
+                android:layout_marginHorizontal="16dp"
+                android:background="?android:selectableItemBackground"
+                app:drawableStartCompat="@drawable/outline_volunteer_activism_24"
+                app:drawableTint="?attr/secondary_color" />
+
+            <LinearLayout
+                android:id="@+id/donationInfoContainer"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:orientation="vertical">
+
+                <LinearLayout
+                    android:id="@+id/lastDonationContainer"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:orientation="horizontal"
+                    android:layout_marginHorizontal="16dp"
```

Comment:
Can we use `padding` instead of `margin` for these clickable elements, so that the ripple effect can extend to the full width, instead of cutting off at the margin?

Reply:
Yes, done!